File: DSAlgorithms/Leetcode/2_pointers/assorted/c++/1768-merge_strings.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
       int i = 0;
       int j = 0; 
        std::string result;
        while(i < word1.size() && j < word2.size())
        {
            result.push_back(word1[i]);
            result.push_back(word2[j]);
            i++;
            j++;
        }
        result.append(word1.substr(i));
        result.append(word2.substr(j));
        return result;
    }
};

File: dsa_algo/leetcode/ts/src/2_pointers/1768-merge_strings.ts
const mergeAlternately = (word1: string, word2: string): string =>{
    let i = 0;
    let j = 0;
    let result: string = "";
    while(i < word1.length && j < word2.length)
    {
        result += word1[i];
        result += word2[j];
        i++;
        j++;
    }
    result += word1.substring(i);
    result += word2.substring(j);
    return result;
};